src/chainstate/stacks/boot/mod.rs

```diff
@@ -249,39 +396,85 @@ impl StacksChainState {
     ///   are repeated floor(stacked_amt / threshold) times.
     /// If an address appears in `addresses` multiple times, then the address's associated amounts
     ///   are summed.
-    pub fn make_reward_set(
-        threshold: u128,
-        mut addresses: Vec<(PoxAddress, u128)>,
-    ) -> Vec<PoxAddress> {
+    pub fn make_reward_set(threshold: u128, mut addresses: Vec<RawRewardSetEntry>) -> RewardSet {
         let mut reward_set = vec![];
+        let mut missed_slots = vec![];
         // the way that we sum addresses relies on sorting.
-        addresses.sort_by_key(|k| k.0.bytes());
-        while let Some((address, mut stacked_amt)) = addresses.pop() {
+        addresses.sort_by_key(|k| k.reward_address.bytes());
+        while let Some(RawRewardSetEntry {
+            reward_address: address,
+            amount_stacked: mut stacked_amt,
+            stacker,
+        }) = addresses.pop()
+        {
+            let mut contributed_stackers = vec![];
+            if let Some(stacker) = stacker.as_ref() {
+                contributed_stackers.push((stacker.clone(), stacked_amt));
+            }
             // peak at the next address in the set, and see if we need to sum
-            while addresses.last().map(|x| &x.0) == Some(&address) {
-                let (_, additional_amt) = addresses
+            while addresses.last().map(|x| &x.reward_address) == Some(&address) {
+                let next_contrib = addresses
                     .pop()
                     .expect("BUG: first() returned some, but pop() is none.");
+                let additional_amt = next_contrib.amount_stacked;
+
+                if let Some(stacker) = next_contrib.stacker {
+                    contributed_stackers.push((stacker.clone(), additional_amt));
+                }
+
                 stacked_amt = stacked_amt
                     .checked_add(additional_amt)
                     .expect("CORRUPTION: Stacker stacked > u128 max amount");
             }
             let slots_taken = u32::try_from(stacked_amt / threshold)
                 .expect("CORRUPTION: Stacker claimed > u32::max() reward slots");
             info!(
-                "Slots taken by {} = {}, on stacked_amt = {}, threshold = {}",
-                &address.clone(),
-                slots_taken,
-                stacked_amt,
-                threshold
+                "Reward slots taken";
+                "reward_address" => %address,
+                "slots_taken" => slots_taken,
+                "stacked_amt" => stacked_amt,
+                "pox_threshold" => threshold,
             );
             for _i in 0..slots_taken {
                 test_debug!("Add to PoX reward set: {:?}", &address);
                 reward_set.push(address.clone());
             }
+            // if stacker did not qualify for a slot *and* they have a stacker
+            //   pointer set by the PoX contract, then add them to auto-unlock list
```

Comment:
The case where they wouldn't have a stacker pointer is if they were in a pool, right?

Reply:
Yep -- if they used delegation.